src/test/java/com/google/sps/data/DatastoreScheduledInterviewTest.java

```diff
@@ -0,0 +1,180 @@
+    tester.create(scheduledInterview3);
+    List<ScheduledInterview> result = tester.getForPerson("user@company.org");
+    ScheduledInterview copyScheduledInterview1 =
+        ScheduledInterview.create(
+            result.get(0).id(),
+            new TimeRange(
+                Instant.parse("2020-07-06T17:00:10Z"), Instant.parse("2020-07-06T18:00:10Z")),
+            "user@company.org",
+            "user@mail.com");
+    ScheduledInterview copyScheduledInterview2 =
+        ScheduledInterview.create(
+            result.get(1).id(),
+            new TimeRange(
+                Instant.parse("2020-07-06T19:00:10Z"), Instant.parse("2020-07-06T20:00:10Z")),
+            "user@company.org",
+            "user2@mail.com");
+    List<ScheduledInterview> expected = new ArrayList<ScheduledInterview>();
+    expected.add(copyScheduledInterview1);
+    expected.add(copyScheduledInterview2);
+    Assert.assertEquals(expected, result);
+  }
+
+  // Tests deleting a user's scheduledInterview.
+  @Test
+  public void deletesScheduledInterview() {
+    tester.create(scheduledInterview1);
+    tester.create(scheduledInterview2);
+    List<ScheduledInterview> result = tester.getForPerson("user@company.org");
+    tester.delete(result.get(0).id());
+    Entity entity = datastore.prepare(new Query("ScheduledInterview")).asSingleEntity();
+    ScheduledInterview storedScheduledInterview = tester.entityToScheduledInterview(entity);
+    ScheduledInterview copyScheduledInterview2 =
+        ScheduledInterview.create(
+            storedScheduledInterview.id(),
+            new TimeRange(
+                Instant.parse("2020-07-06T19:00:10Z"), Instant.parse("2020-07-06T20:00:10Z")),
+            "user@company.org",
+            "user2@mail.com");
+    Assert.assertEquals(copyScheduledInterview2, storedScheduledInterview);
+  }
+
+  // Tests updating a user's scheduledInterview.
+  @Test
+  public void updatesScheduledInterview() {
+    tester.create(scheduledInterview1);
+    Entity entity = datastore.prepare(new Query("ScheduledInterview")).asSingleEntity();
+    ScheduledInterview previousStoredScheduledInterview = tester.entityToScheduledInterview(entity);
+    ScheduledInterview updatedStoredScheduledInterview =
+        ScheduledInterview.create(
+            previousStoredScheduledInterview.id(),
+            new TimeRange(
+                Instant.parse("2020-07-06T19:00:10Z"), Instant.parse("2020-07-06T20:00:10Z")),
+            "user@company.org",
+            "user3@mail.com");
+    tester.update(updatedStoredScheduledInterview);
+    Entity updatedEntity = datastore.prepare(new Query("ScheduledInterview")).asSingleEntity();
+    ScheduledInterview updatedScheduledInterview = tester.entityToScheduledInterview(updatedEntity);
+    Assert.assertEquals(updatedStoredScheduledInterview, updatedScheduledInterview);
+  }
+
+  // Tests retrieving a scheduledInterview from Datastore.
+  @Test
+  public void getsScheduledInterview() {
```

Comment:
You should add a test where it is not within datastore to make sure an empty Optional is returned.

Reply:
Done